tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ManagedExecutorTest.java

```diff
@@ -481,6 +485,207 @@ public void failedStageDependentStagesRunWithContext() throws ExecutionException
+     */
+    @Test
+    public void maxQueued3() throws ExecutionException, InterruptedException, TimeoutException {
+        ManagedExecutor executor = ManagedExecutor.builder()
+                .maxAsync(1)
+                .maxQueued(3)
+                .build();
+
+        try {
+            // First, use up the single maxAsync slot with a blocking task and wait for it to start
+            Phaser barrier = new Phaser(1);
+            executor.submit(() -> barrier.awaitAdvanceInterruptibly(barrier.arrive() + 1));
+            barrier.awaitAdvanceInterruptibly(0, MAX_WAIT_NS, TimeUnit.NANOSECONDS);
+
+            // Use up first queue position
+            Future<Integer> future1 = executor.submit(() -> 101);
+
+            // Use up second queue position
+            CompletableFuture<Void> future2 = executor.runAsync(() -> System.out.println("second task running"));
+
+            // Use up third queue position
+            Future<Integer> future3 = executor.submit(() -> 103);
+
+            // Fourth attempt to queue a task must be rejected
+            try {
+                Future<Integer> future4 = executor.submit(() -> 104);
+                Assert.fail("Exceeded maxQueued of 3. Future for 4th queued task/action is " + future4);
+            }
+            catch (RejectedExecutionException x) {
+                // test passes
+            }
+
+            // Fifth attempt to queue a task must also be rejected
+            try {
+                CompletableFuture<Integer> future5 = executor.supplyAsync(() -> 105);
+                Assert.fail("Exceeded maxQueued of 3. Future for 5th queued task/action is " + future5);
+            }
+            catch (RejectedExecutionException x) {
+                // test passes
+            }
+
+            // unblock and allow tasks to finish
+            barrier.arrive();
+
+            Assert.assertEquals(future1.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(101),
+                    "Unexpected result of first task.");
+
+            Assert.assertNull(future2.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS),
+                    "Unexpected result of second task.");
+
+            // At least 2 queue positions must be available at this point
+            Future<Integer> future6 = executor.submit(() -> 106);
+            CompletableFuture<Integer> future7 = executor.supplyAsync(() -> 107);
+
+            Assert.assertEquals(future3.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(103),
+                    "Unexpected result of third task.");
+
+            Assert.assertEquals(future6.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(106),
+                    "Unexpected result of sixth task.");
+
+            Assert.assertEquals(future7.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(107),
+                    "Unexpected result of seventh task.");
+        }
+        finally {
+            executor.shutdownNow();
```

Comment:
do we need to have a `barrier.forceTermination()` here too?

Reply:
I added it on the other test because the async tasks there do blocking without timeouts.  However, it wouldn't hurt to have it here as well, and as long as I'm making the other requested change anyway, I'll add it.